activerecord/test/cases/associations/has_one_associations_test.rb

```diff
@@ -941,3 +941,37 @@ def test_has_one_with_touch_option_on_nonpersisted_built_associations_doesnt_upd
     MESSAGE
   end
 end
+
+class AsyncHasOneAssociationsTest < ActiveRecord::TestCase
+  include WaitForAsyncTestHelper
+
+  fixtures :companies, :accounts
+
+  self.use_transactional_tests = false
```

Comment:
Note to self. With the recent refactor in connection pool / pinned connections, we should actually be able to properly simulate async queries with transactional tests. We just need a lock around the connection.